Cloud Architecture Design Guidance (#963)
microsoft/code-with-engineering-playbook: docs/design/design-patterns/network-architecture-guidance-for-azure/readme.md

Implementing a Hub-and-Spoke model in Azure offers several benefits:

* Isolation and Segmentation: By dividing resources into separate spoke VNets, you can isolate and segment workloads, preventing any potential issues or security risks from affecting other parts of the network.
* Centralized Management: The hub VNet acts as a single point of management for shared services, making it easier to maintain, monitor, and enforce policies across the network.
* Simplified Connectivity: VNet peering enables seamless communication between the hub and spoke VNets without the need for complex routing or additional gateways, reducing latency and management overhead.
* Scalability: The Hub-and-Spoke model can easily scale to accommodate additional spokes as the organization grows or as new applications and services are introduced.
* Cost Savings: By centralizing shared services in the hub, organizations can reduce the costs associated with deploying and managing multiple instances of the same services across different VNets.

> Read more about [Hub-And-Spoke topology](https://learn.microsoft.com/en-us/azure/architecture/reference-architectures/hybrid-networking/hub-spoke?tabs=cli)

Reviewer: ```suggestion
Read more about [Hub-And-Spoke topology](https://learn.microsoft.com/azure/architecture/reference-architectures/hybrid-networking/hub-spoke?tabs=cli)
```

> When deploying hub/spoke, it is recommended that you do so in connection with [landing zones](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/landing-zone/). This ensures consistency across all environments as well as guardrails to ensure a high level of security while giving developers freedom within development environments.

Reviewer: ```suggestion
When deploying hub/spoke, it is recommended that you do so in connection with [landing zones](https://learn.microsoft.com/azure/cloud-adoption-framework/ready/landing-zone/). This ensures consistency across all environments as well as guardrails to ensure a high level of security while giving developers freedom within development environments.
```

### Firewall and security

When using a Hub-and-Spoke topology it is possible to deploy a centralized Firewall instance in the Hub that all outgoing traffic or traffic to/from certain VNets, this allows for centralized threat protection while minimizing costs.

### DNS

> The best practices for handling DNS in azure and in clouds in general include using the managed DNS services. Some of the benefits of using the managed DNS services is that the resources are designed to be secure, and easy to deploy and configure.

Reviewer: ```suggestion
The best practices for handling DNS in Azure, and in cloud environments in general, include using managed DNS services. Some of the benefits of using managed DNS services is that the resources are designed to be secure, easy to deploy and configure.
```

* **DNS forwarding:** Set up DNS forwarding between your on-premises DNS servers and Azure DNS servers for name resolution across environments.
* **Use Azure Private DNS zones for Azure resources:** Configure Azure Private DNS zones for your Azure resources to ensure name resolution is kept within the virtual network.

> Read more about [Hybrid/Multi-Cloud DNS infrastructure](https://learn.microsoft.com/en-us/azure/architecture/hybrid/hybrid-dns-infra) and [Azure DNS infratructure](https://learn.microsoft.com/en-us/azure/dns/)

Reviewer: ```suggestion
Read more about [Hybrid/Multi-Cloud DNS infrastructure](https://learn.microsoft.com/azure/architecture/hybrid/hybrid-dns-infra) and [Azure DNS infratructure](https://learn.microsoft.com/azure/dns/)
```

* **Use subnets:** Divide your VNets into smaller subnets based on security, application, or environment requirements. This allows for better network management and security.
* **Consider leaving a buffer between VNets:** While it's not strictly necessary, leaving a buffer between VNets can be beneficial in some cases, especially when you anticipate future growth or when you might need to merge VNets. This can help avoid re-addressing conflicts when expanding or merging networks.
* **Reserve IP addresses:** Reserve a range of IP addresses within your VNet address space for critical resources or services. This ensures that they have a static IP address, which is essential for specific services or applications.
* **Plan for hybrid scenarios:** If you're working in a hybrid environment with on-premises or multi-cloud networks, ensure that you plan for IP address allocation across all environments. This includes avoiding overlapping address spaces and reserving IP addresses for specific resources like VPN gateways or ExpressRoute circuits.

> Read more at [azure-best-practices/plan-for-ip-addressing](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/plan-for-ip-addressing)

Reviewer: ```suggestion
Read more at [azure-best-practices/plan-for-ip-addressing](https://learn.microsoft.com/azure/cloud-adoption-framework/ready/azure-best-practices/plan-for-ip-addressing)
```
